src/components/cc-domain-management/cc-domain-management.js

```diff
@@ -0,0 +1,819 @@
+ */
+
+/**
+ * A component to manage domains associated to an application.
+ *
+ * @cssdisplay block
+ *
+ * @fires {CustomEvent<{ hostname: string, pathPrefix: string }>} cc-domain-management:add - Fires when clicking the "Add a domain" button.
+ * @fires {CustomEvent<DomainInfo>} cc-domain-management:mark-as-primary - Fires when clicking a "mark as primary" button.
+ * @fires {CustomEvent<DomainInfo>} cc-domain-management:delete - Fires when clicking a "delete" button.
+ *
+ */
+export class CcDomainManagement extends LitElement {
+
+  static get properties () {
+    return {
+      dnsInfoState: { type: Object, attribute: 'dns-info-state' },
+      domainFormState: { type: Object, attribute: 'domain-form-state' },
+      domainListState: { type: Object, attribute: 'domain-list-state' },
+      _sortedDomains: { type: Array, state: true },
+    };
+  }
+
+  /**
+   * @returns {DomainManagementFormStateIdle}
+   */
+  static get INIT_DOMAIN_FORM_STATE () {
+    return {
+      type: 'idle',
+      hostname: {
+        value: '',
+      },
+      pathPrefix: {
+        value: '',
+      },
+    };
+  }
+
+  constructor () {
+    super();
+
+    /** @type {DomainManagementDnsInfoState} Sets the state of the DNS info section */
+    this.dnsInfoState = { type: 'loading' };
+
+    /** @type {DomainManagementFormState} Sets the state of the form section */
+    this.domainFormState = CcDomainManagement.INIT_DOMAIN_FORM_STATE;
+
+    /** @type {DomainManagementListState} Sets the state of the domain list section. */
+    this.domainListState = { type: 'loading' };
+
+    /** @type {RefCcInputText} */
+    this._domainInputRef = createRef();
+
+    /** @type {RefHTMLParagraphElement} */
+    this._emptyMessageRef = createRef();
+
+    /** @type {FormattedDomainInfo[]|null} */
+    this._sortedDomains = null;
+
+    new LostFocusController(this, '.delete-domain', ({ suggestedElement }) => {
+      if (suggestedElement != null) {
+        suggestedElement.focus();
+      }
+      else {
+        this._emptyMessageRef.value?.focus();
+      }
+    });
+
+    new LostFocusController(this, '.mark-primary', () => {
+      focusBySelector(this, '.primary');
+    });
+  }
+
+  /**
+   * @param {string} hostnameValue
+   * @returns {FormError|null} The error code or `null` if no error
+   */
+  _getErrorCode (hostnameValue) {
+    try {
+      const { pathname } = parseDomain(hostnameValue?.trim());
+
+      if (!hostnameValue.match(/.+\..+/)) {
+        return 'invalid-format';
+      }
+
+      if (pathname !== '/') {
+        /** We store the path so we can tell users to move this part to the "" input field instead. */
+        return { code: 'hostname-contains-path', pathWithinHostname: pathname };
+      }
+
+      return null;
+    }
+    catch (error) {
+      if (error instanceof DomainError) {
+        return error.message;
+      }
+
+      return 'invalid-format';
+    }
+
+  }
```

Comment:
Dicussed in Sync with Pierre (thanks again ❤️)
The parseDomain helper throws errors when the domain to parse:

    is empty,
    has invalid wildcard (toto*.example.com is invalid, "*" may only be used as a subdomain, it cannot be part of a subdomain)
    has invalid format.

This is why we use try / catch here.

The hostname-contains-path is not part of what the parseDomain checks because it's very specific.
It's only a problem within this component and only when users try to enter a path within the domain input field. In other contexts, domains containing a path are fine since it's a feature Clever Cloud supports.